libraries/AP_IRLock/pixy_parser.h

```diff
@@ -0,0 +1,84 @@
+
+/*
+ * pixy_parser.h
+ *
+ */
+
+#pragma once
+
+#include <ctype.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <string.h>
+
+#define PIXY_PARSER_PIXY_BUF_SIZE 17
+
+/*
+    pixy_buf_size = 17;
+    pixy_buf[17]= {};
+    pixy_len = 0;
+    blob_buffer_write_idx = 0;
+    bytes_to_sof = 0;
+    bytes_to_block = 0;
+}
+*/
+class pixy_parser {
+    public:
+        bool if_swap_buffer;
```

Comment:
Avoid public member variables. I recommend making recv_byte_pixy() return true when the buffer is swapped.

Reply:
Done